zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java

```diff
@@ -85,6 +86,14 @@ public static Path path(final URI uri) {
    * <p>NOTE: Zoltar internal use only!</p>
    */
   public static URI downloadIfNonLocal(final URI path) throws IOException {
+    // If the URI is a jar file, register it as a file system so Paths.get is able to access the
+    // files in it. We then copy the files into a temp directory because the files must be in a
+    // normal directory so Tensorflow can read them. See this Oracle doc for more details:
+    // https://docs.oracle.com/javase/7/docs/technotes/guides/io/fsp/zipfilesystemprovider.html
+    if ("jar".equals(path.getScheme())) {
+      FileSystems.newFileSystem(path, ImmutableMap.of());
```

Comment:
Sure, if you already have this working we can close this PR and use yours instead. Would you be able to create a new PR?

Reply:
@regadas I took a closer look at your implementation and had a couple of questions. I agree it would be nice to avoid copying the files, but doesn't Tensorflow require that the files be in a regular directory on disk? Is it possible for Tensorflow to read the model directly from the jar? I tried passing Tensorflow a JAR URI (e.g. `jar:file:/...`) but that didn't work.

And maybe I'm missing something, but it looks to me like your implementation also make a copy? The `if` statement on [line 110](https://github.com/regadas/zoltar/blob/feature/bazel/zoltar-core/src/main/java/com/spotify/zoltar/fs/FileSystemExtras.java#L110) will be false since the ZipFS won't be equal to the default file system, so it would copy the files to a temp directory?

I like that your implementation registers the new filesystem in the `path` method, so I changed this PR to do the same.

Reply:
@rculbertson yeah nice catch! yes TF `SavedModelBundle` native api expects an unpacked dir, I must have confused myself with the graph api which would allow us to do this.

Let's merge this and keep it simple for now ... I'll just update one little thing to avoid registering the FS every time.